File: ProgrammingProblemSolutions/ProgrammingProblemCode/ToLowerCaseProcessor.cs
namespace ProgrammingProblemSolutions.ProgrammingProblemCode
{
    public class ToLowerCaseProcessor
    {
        //Faster than 95.89%
        public string ToLowerCase(string str)
        {
            string result = string.Empty;
            foreach (char c in str)
            {
                if (c >= 65 && c <= 90)
                {
                    result += (char)(c + 32);
                }
                else
                {
                    result += c;
                }
            }

            return result;
        }
    }
}

File: PythonProgrammingProblemSolutions/ProgrammingProblemCode/ToLowerCaseProcessor.py
# tolowercaseprocessor.py file
# faster than 99.74%, memory usage less than 96.86%

class ToLowerCaseProcessor(object):
    def toLowerCase(self, str: 'str') -> 'str':
        result = map(lambda x:chr(ord(x)+32) if 90>=ord(x)>=65 else x,list(str))
        return "".join(result)